PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ['maintenance.equipment']
-    _inherit = ['maintenance.request']
-    _inherit = ['res.users']
-    _inherit = ['hr.employee']

Total Changes: 8

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_maintenance/models/equipment.py
+++ b/addons/hr_maintenance/models/equipment.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models, tools
 
 
 class MaintenanceEquipment(models.Model):
-    _inherit = ['maintenance.equipment']
+    _inherit = 'maintenance.equipment'
 
     employee_id = fields.Many2one('hr.employee', compute='_compute_equipment_assign',
         store=True, readonly=False, string='Assigned Employee', tracking=True)
@@ -78,7 +78,7 @@ class MaintenanceEquipment(models.Model):
 
 
 class MaintenanceRequest(models.Model):
-    _inherit = ['maintenance.request']
+    _inherit = 'maintenance.request'
 
     @api.returns('self')
     def _default_employee_get(self):

--- a/addons/hr_maintenance/models/res_users.py
+++ b/addons/hr_maintenance/models/res_users.py
@@ -2,7 +2,7 @@ from odoo import api, models, fields
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     equipment_ids = fields.One2many('maintenance.equipment', 'owner_user_id', string="Managed Equipment")
     equipment_count = fields.Integer(related='employee_id.equipment_count', string="Assigned Equipment")
@@ -13,7 +13,7 @@ class ResUsers(models.Model):
 
 
 class HrEmployee(models.Model):
-    _inherit = ['hr.employee']
+    _inherit = 'hr.employee'
 
     equipment_ids = fields.One2many('maintenance.equipment', 'employee_id', groups="hr.group_hr_user")
     equipment_count = fields.Integer('Equipment Count', compute='_compute_equipment_count', groups="hr.group_hr_user")
